portfolio/src/main/java/com/google/sps/data/MemePost.java

```diff
@@ -0,0 +1,82 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.util.Date;
+
+/**
+ * The MemePost class holds data for the user's MemePost
+ */
+public class MemePost{
+    
+    private String author;
+    private String description;
+    private String imageUrl;
+    private long likes;
+    private long dislikes;
+    // private Date date;
+    
+    public MemePost(String author, String description, String imageUrl){
+        this.author = author;
+        this.description = description;
+        this.imageUrl = imageUrl;
+        this.likes = 0;
+        this.dislikes = 0;
+        // this.Date = new Date();
+    }
+
+    public MemePost(String author, String description, String imageUrl, long likes, long dislikes, Date date){
+        this.author = author;
+        this.description = description;
+        this.imageUrl = imageUrl;
+        this.likes = likes;
+        this.dislikes = dislikes;
+        // this.Date = date;
+    }
+
+    public String getAuthor(){
+        return this.author;
+    }
+
+    public void setAuthor(String author){
+        this.author = author;
+    }
+
+    public String getDescription(){
+        return this.description;
+    }
+
+    public void setDescription(String description){
+        this.description = description;
+    }
+
+    public void increaseLikes() {
+        this.likes++;
+    }
+
+    public void decreaseLikes() {
+        this.likes--;
+    }
+
+    public void increaseDislikes() {
+        this.dislikes++;
+    }
+
+    public void decreaseDislikes() {
+        this.dislikes--;
+    }
+
+    public void putInDatastore(){
```

Comment:
This should probably not use the DatastoreService itself, but return the Entity, and allow callers to do whatever they want with that entity. (For example what if I have ten instances of the MemePost object and I want to put a "dank memes" list into the datastore? I want to get the database entity, but I want to do my own thing with it instead of just a `put`)

Reply:
This seems a lot more efficient. Will do.